components/suggestion/switch_channel_provider.jsx

```diff
@@ -218,16 +212,40 @@ function quickSwitchSorter(wrappedA, wrappedB) {
 
     const aStartsWith = aDisplayName.startsWith(prefix);
     const bStartsWith = bDisplayName.startsWith(prefix);
-    if ((aStartsWith && bStartsWith) || (!aStartsWith && !bStartsWith)) {
-        //
-        // MM-12677 When this is migrated this needs to be fixed to pull the user's locale
-        //
+
+    // Open channels user havent interacted should be at the  bottom of the list
+    if (a.type === Constants.OPEN_CHANNEL && !wrappedA.last_viewed_at && (b.type !== Constants.OPEN_CHANNEL || wrappedB.last_viewed_at)) {
+        return 1;
+    } else if (b.type === Constants.OPEN_CHANNEL && !wrappedB.last_viewed_at) {
+        return -1;
+    }
+
+    // If both channels start with search term then sort by last_viewed_at
+    if (aStartsWith && bStartsWith) {
+        if (wrappedA.last_viewed_at && wrappedB.last_viewed_at) {
+            return wrappedB.last_viewed_at - wrappedA.last_viewed_at;
+        } else if (wrappedA.last_viewed_at) {
+            return -1;
+        } else if (wrappedB.last_viewed_at) {
+            return 1;
+        }
         return sortChannelsByTypeAndDisplayName('en', a, b);
-    } else if (aStartsWith) {
+    }
+
+    if (aStartsWith) {
+        return -1;
+    } else if (bStartsWith) {
+        return 1;
+    } else if (wrappedA.last_viewed_at && !wrappedB.last_viewed_at) {
         return -1;
+    } else if (!wrappedA.last_viewed_at && wrappedB.last_viewed_at) {
+        return 1;
+    } else if (wrappedA.last_viewed_at && wrappedB.last_viewed_at) {
+        return wrappedB.last_viewed_at - wrappedA.last_viewed_at;
```

Comment:
Could this maybe be grouped differently to make it easier to follow? Something like
```javascript
// Sort channels starting with the search term first
if (aStartsWith && !bStartsWith) {
    return -1;
} else if (!aStartsWith && bStartsWith) {
    return 1;
}

// At this point, either both channels start with the term or neither channel does

// Sort recently viewed channels first
if (wrappedA.last_viewed_at && wrappedB.last_viewed_at) {
    return wrappedB.last_viewed_at - wrappedA.last_viewed_at;
} else if (wrappedA.last_viewed_at) {
    return -1;
} else if (wrappedB.last_viewed_at) {
    return 1;
}

// Similarly, at this point, neither channel has been previously viewed before

return sortChannelsByTypeAndDisplayName('en', a, b);
```
That saves us from repeating ourselves a bit and it keeps the different conditions separate.

Reply:
i can check but some of the test cases are failing.

Reply:
Awesome, Thank you. This is much cleaner